galsim/compound.py

```diff
@@ -645,6 +744,33 @@ def __str__(self):
         s += ')'
         return s
 
+    def _prepareDraw(self):
+        self._orig_obj._prepareDraw()
+        self.SBProfile = galsim._galsim.SBAutoConvolve(self._orig_obj.SBProfile, self._real_space,
+                                                       self._gsparams)
+
+    def shoot(self, n_photons, rng=None):
+        """Shoot photons into a PhotonArray.
+
+        @param n_photons    The number of photons to use for photon shooting.
+        @param rng          If provided, a random number generator to use for photon shooting,
+                            which may be any kind of BaseDeviate object.  If `rng` is None, one
+                            will be automatically created, using the time as a seed.
+                            [default: None]
+        @returns PhotonArray.
+        """
+        # Setup the rng if not provided one.
+        if rng is None:
```

Comment:
Sure.  Actually, is there any reason why we shouldn't make it so this whole snippet:
```python
    if rng is None:
        ud = galsim.UniformDeviate()
    elif isinstance(rng, galsim.BaseDeviate):
        ud = galsim.UniformDeviate(rng)
    else:
        raise TypeError("The rng provided is not a BaseDeviate")
```

is equivalent to just

```python
ud = galsim.UniformDeviate(rng)
```
??